src/extension/background-script/connectors/alby.ts

```diff
@@ -0,0 +1,620 @@
+            const token = await authClient.refreshAccessToken();
+            if (authClient.token) {
+              authClient.token.access_token = token.token.access_token;
+              authClient.token.refresh_token = token.token.refresh_token;
+            }
+          } catch {
+            const authResult = await this.launchWebAuthFlow(lnurlAuthUrl);
+
+            if (authResult) {
+              await this.getAccessToken(authResult, authClient);
+            } else {
+              throw new Error("Authentication failed: missing authResult");
+            }
+          }
+
+          const account = await get({
+            action: "getAccount",
+            origin: { internal: true },
+          });
+
+          if (account) {
+            edit({
+              action: "editAccount",
+              args: {
+                name: account.data?.name,
+                id: account.data?.id,
+                accessToken: authClient.token?.access_token,
+                refreshToken: authClient.token?.refresh_token,
+                tokenExpiresAt: authClient.token?.expires_at,
+              },
+              origin: {
+                internal: true,
+              },
+            });
+          }
+        }
+      } else {
+        const authResult = await this.launchWebAuthFlow(lnurlAuthUrl);
+
+        if (authResult) {
+          await this.getAccessToken(authResult, authClient);
+        } else {
+          throw new Error("Authentication failed: missing authResult");
+        }
+      }
+
+      this.access_token = authClient.token?.access_token;
+      const expires_at = authClient.token?.expires_at;
+
+      this.access_token_created = +new Date();
+      this.refresh_token = authClient.token?.refresh_token;
+
+      this.refresh_token_created = +new Date();
+
+      const authData: {
+        access_token?: string;
+        refresh_token?: string;
+        expires_at?: number;
+      } = {};
+
+      authData.access_token = this.access_token;
+
+      authData.refresh_token = this.refresh_token;
+      authData.expires_at = expires_at;
+
+      return authData;
+    } catch (e) {
+      throw new Error(
+        `API error (${this.config.url})${
+          e instanceof Error ? `: ${e.message}` : ""
+        }`
+      );
+    }
+  }
+
+  async launchWebAuthFlow(lnurlAuthUrl: string) {
+    const authResult = await browser.identity.launchWebAuthFlow({
+      interactive: true,
+      url: lnurlAuthUrl,
+    });
+
+    return authResult;
+  }
+
+  async getAccessToken(authResult: string, authClient: auth.OAuth2User) {
+    const authToken = new URL(authResult).searchParams.get("code");
+    if (!authToken) {
+      throw new Error("Authentication failed: missing token");
+    }
+
+    await authClient.requestAccessToken(authToken);
+  }
+
+  generateHmacVerification(uri: string) {
+    const mac = hmacSHA256(uri, HMAC_VERIFY_HEADER_KEY).toString(Base64);
+    return encodeURIComponent(mac);
+  }
+
+  async request<Type>(
```

Comment:
Why not use the alby-js-sdk client? otherwise we need to reimplement all the calls here? e.g. just use `const result = await client.sendPayment({ invoice: bolt11 })`; instead of doing the http call manually.